File: app-java/src/main/java/com/TritiumGaming/phasmophobiaevidencepicker/activities/mainmenus/newsletter/NewsMessageFragment.java
package com.TritiumGaming.phasmophobiaevidencepicker.activities.mainmenus.newsletter;

import android.os.Bundle;
import android.text.Html;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.widget.AppCompatTextView;
import androidx.navigation.Navigation;

import com.TritiumGaming.phasmophobiaevidencepicker.activities.mainmenus.newsletter.data.NewsletterMessageData;
import com.TritiumGaming.phasmophobiaevidencepicker.R;
import com.TritiumGaming.phasmophobiaevidencepicker.activities.mainmenus.MainMenuFragment;
import com.TritiumGaming.phasmophobiaevidencepicker.views.PETImageButton;

/**
 * TitleScreenFragment class
 *
 * @author TritiumGamingStudios
 */
public class NewsMessageFragment extends MainMenuFragment {

    @Override
    public void onCreate(@Nullable @org.jetbrains.annotations.Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Nullable
    @Override
    public View onCreateView(
            @NonNull LayoutInflater inflater,
            @Nullable ViewGroup container,
            @Nullable Bundle savedInstanceState) { // OBTAIN VIEW MODEL REFERENCE

        super.init();

        return inflater.inflate(R.layout.fragment_news_message, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {

        PETImageButton button_back = view.findViewById(R.id.button_left);
        // INITIALIZE VIEWS
        AppCompatTextView label_title = view.findViewById(R.id.textview_messageTitle);
        AppCompatTextView label_date = view.findViewById(R.id.textView_messageDate);
        AppCompatTextView label_content = view.findViewById(R.id.textView_messageContent);

        // LISTENERS
        button_back.setOnClickListener(v -> Navigation.findNavController(v).popBackStack());

        // SET CONTENT
        NewsletterMessageData message = newsLetterViewModel.getCurrentMessage();
        if (message != null) {
            label_title.setText(Html.fromHtml(message.getTitle()));
            label_date.setText(Html.fromHtml(message.getDate()));
            label_content.setText(Html.fromHtml(message.getDescription()));
        } else {
            label_title.setText(Html.fromHtml("Data unavailable"));
            label_date.setText(Html.fromHtml("Data unavailable"));
            label_content.setText(Html.fromHtml("Data unavailable"));
        }

        super.initAdView(view.findViewById(R.id.adView));
    }

    @Override
    protected void initViewModels() {
        super.initViewModels();
        initMainMenuViewModel();
        initNewsletterViewModel();
    }

}

File: app-java/src/main/java/com/TritiumGaming/phasmophobiaevidencepicker/activities/mainmenus/newsletter/NewsMessageFragment.kt
package com.TritiumGaming.phasmophobiaevidencepicker.activities.mainmenus.newsletter

import android.os.Bundle
import android.text.Html
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.appcompat.widget.AppCompatTextView
import androidx.navigation.Navigation.findNavController
import com.TritiumGaming.phasmophobiaevidencepicker.R
import com.TritiumGaming.phasmophobiaevidencepicker.activities.mainmenus.MainMenuFragment
import com.TritiumGaming.phasmophobiaevidencepicker.views.global.PETImageButton

/**
 * TitleScreenFragment class
 *
 * @author TritiumGamingStudios
 */
class NewsMessageFragment : MainMenuFragment() {

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? { // OBTAIN VIEW MODEL REFERENCE

        super.init()

        return inflater.inflate(R.layout.fragment_news_message, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        val buttonBack = view.findViewById<PETImageButton>(R.id.button_left)
        // INITIALIZE VIEWS
        val labelTitle = view.findViewById<AppCompatTextView>(R.id.textview_messageTitle)
        val labelDate = view.findViewById<AppCompatTextView>(R.id.textView_messageDate)
        val labelContent = view.findViewById<AppCompatTextView>(R.id.textView_messageContent)

        // LISTENERS
        buttonBack.setOnClickListener { v: View? ->
            v?.let { findNavController(v).popBackStack() }
        }

        // SET CONTENT
        val message = newsLetterViewModel?.currentMessage
        message?.let {
            labelTitle.text = Html.fromHtml(message.title)
            labelDate.text = Html.fromHtml(message.date)
            labelContent.text = Html.fromHtml(message.description)
        } ?: {
            labelTitle.text = Html.fromHtml("Data unavailable")
            labelDate.text = Html.fromHtml("Data unavailable")
            labelContent.text = Html.fromHtml("Data unavailable")
        }

        super.initAdView(view.findViewById(R.id.adView))
    }

    override fun initViewModels() {
        super.initViewModels()
        initMainMenuViewModel()
        initNewsletterViewModel()
    }
}